crates/runtime/src/flight.rs

```diff
@@ -229,6 +233,38 @@ impl Service {
     }
 }
 
+fn verify_schema(
```

Comment:
Schema already implements `Eq` - so you can just compare two schemas with `==`

i.e. 
```rust
let schema1 = arrow::datatypes::Schema::new(vec![
    arrow::datatypes::Field::new("a", arrow::datatypes::DataType::Int64, false),
    arrow::datatypes::Field::new("b", arrow::datatypes::DataType::Utf8, false),
]);

let schema2 = arrow::datatypes::Schema::new(vec![
    arrow::datatypes::Field::new("a", arrow::datatypes::DataType::Int64, false),
    arrow::datatypes::Field::new("b", arrow::datatypes::DataType::Utf8, false),
]);

assert_eq!(schema1, schema2); // True!
if schema1 == schema2 {
  println!("Also true!");
}
```

Reply:
If we need a more lenient schema comparison, then we can use one of the two methods in https://docs.rs/datafusion/latest/datafusion/common/trait.SchemaExt.html#required-methods
`equivalent_names_and_types` or `logically_equivalent_names_and_types`

Reply:
Ok that makes sense